packages/block-library/src/table/edit.js

```diff
@@ -355,6 +361,30 @@ function TableEdit( {
 		}
 	}, [ hasTableCreated ] );
 
+	// We need to show the caption when changes come from
+	// history navigation(undo/redo).
+	useEffect( () => {
+		if ( caption && ! prevCaption ) {
+			setShowCaption( true );
+		}
+	}, [ caption, prevCaption ] );
+
+	useEffect( () => {
+		if ( ! isSelected && ! caption ) {
+			setShowCaption( false );
+		}
+	}, [ isSelected, caption ] );
+
+	// Focus the caption when we click to add one.
+	const captionRef = useCallback(
+		( node ) => {
+			if ( node && ! caption ) {
+				node.focus();
```

Comment:
I think there's an a11y bug where undo/redo-ing from an empty caption to some content in the caption will move the focus to the caption `<RichText>`. This might not be ideal when the focus is previously on the undo/redo button, for instance. @alexstine knows more about this than me.

Reply:
I may not know exactly what the problem is, but is it a case of, for example, the following? In both cases, I can confirm that the focus jumps.

<details>
<summary>In the table block, undoing typed text shifts focus to the caption.</summary>

https://user-images.githubusercontent.com/54422211/219870217-a4766ab6-50c4-47ec-bb49-7102b913dfbe.mp4

</details>

<details>
<summary>Undoing on the first deleted paragraph brings the focus to the second paragraph</summary>

https://user-images.githubusercontent.com/54422211/219870302-0707a88e-e7c5-4f86-97f1-089e23d26d77.mp4

</details>

Reply:
Oops! Sorry for the confusion. I meant the undo/redo "buttons" in the toolbar rather than the shortcuts. See the below screencast for example:

https://user-images.githubusercontent.com/7753001/220261241-c5c8fd7c-6c5b-48e2-b3e8-a9f1ac0ecd8b.mp4